src/lib/services/contentful/fetch.test.ts

```diff
@@ -0,0 +1,59 @@
+import { describe, expect, it, vi, type Mock } from "vitest";
+import * as envVars from "$env/static/private";
+import contentfulFetch from "./fetch";
+
+vi.mock("$env/static/private");
+type EnvVarName = "VERCEL" | "CONTENTFUL_SPACE_ID" | "CONTENTFUL_DELIVERY_API_TOKEN";
+type EnvVars = Record<EnvVarName, string>;
+const setEnvVars = (vars: EnvVars) => {
+  let key: keyof EnvVars;
+  for (key in vars) {
+    (envVars as EnvVars)[key] = vars[key];
+  }
+};
```

Comment:
After much floundering with getting a mock working here, I ultimately landed here after shamelessly stealing from @Benaiah's work on the Intersection Observer test: https://github.com/la-ldaf/ldaf-site/blob/1e67dd643afd6accf014629c68cc3696a43416fc/src/lib/components/IntersectionObserver/IntersectionObserver.test.ts#L12-L18

There really doesn't seem to be a great way to mock `$env/static/private` (or any of the other SvelteKit `$env/*` imports), and since `vi.mock` gets hoisted I couldn't really find a cleaner way of doing this. Some other attempts and considerations included:
* Tried using `vi.stubEnv`, which seemed like it would have been perfect, except by the time the tests run it's too late to update env vars (changes don't get reflected in `$env/*`)
* Tried using `vi.doMock` within each test, which didn't work and also didn't seem like the right use-case since we're mocking an import of the file we want to test and not the file itself.
* I would have liked to use something like `vi.fn().mockImplementationOnce()` but `$env/*` doesn't utilize functions for me to mock.
* Considered setting up a different mock of `$env/static/private` in `__mocks__/` for each use-case, which just felt a bit messy, so I didn't attempt it (not even sure if it would have worked).

All that to say, I'm still not super happy with this, so I'm very open to any suggestions on how we can improve it. It does work as-is though, so I'm also fine with moving forward with what we've got.

Reply:
I am never above doing whatever witchcraft is required to make test mocks work as desired 😅

Reply:
I'm fine with this approach! Looks similar to what I did in my intersection observer tests. I think we might be able to re-use this technique for the other file below, going to give this a shot.

Reply:
@Benaiah sounds good! To get this moving along, are you okay with trying that out in a follow-up, or did you want to see about getting that change in here?

Reply:
I'm fine with trying that out in a follow-up and approving this to go in for now, sure!